src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -152,21 +152,27 @@ protected override void ReceiveByteArray(int id, byte[] data)
 
     protected override async Task TransmitStreamAsync(long streamId, DotNetStreamReference dotNetStreamReference)
     {
-        if (!_pendingDotNetToJSStreams.TryAdd(streamId, dotNetStreamReference))
+        var cancelableStreamReference = new CancelableDotNetStreamReference(dotNetStreamReference);
+        if (!_pendingDotNetToJSStreams.TryAdd(streamId, cancelableStreamReference))
         {
             throw new ArgumentException($"The stream {streamId} is already pending.");
         }
 
         // SignalR only supports streaming being initiated from the JS side, so we have to ask it to
         // start the stream. We'll give it a maximum of 10 seconds to do so, after which we give up
         // and discard it.
-        var cancellationToken = new CancellationTokenSource(TimeSpan.FromSeconds(10)).Token;
-        cancellationToken.Register(() =>
+        CancellationTokenSource cancellationTokenSource = new(TimeSpan.FromSeconds(10));
+
+        // Store CTS to dispose later.
+        cancelableStreamReference.CancellationTokenSource = cancellationTokenSource;
+
+        cancellationTokenSource.Token.Register(() =>
         {
             // If by now the stream hasn't been claimed for sending, stop tracking it
-            if (_pendingDotNetToJSStreams.TryRemove(streamId, out var timedOutStream) && !timedOutStream.LeaveOpen)
+            if (_pendingDotNetToJSStreams.TryRemove(streamId, out var timedOutCancelableStreamReference))
```

Comment:
the existing code appears to prefer to check LeaveOpen and dispose the stream directly rather than just calling Dispose (which does the same check).  I'm not familiar with the history but is there a guiding style principle here @javiercn